app/controllers/build_estimates_controller.rb

```diff
@@ -8,9 +8,13 @@ def update
     if @form.valid?
       session_data.merge!(@form.attributes)
       estimate = load_estimate
-      handler.save_data(cfe_connection, estimate_id, @form, session_data) if last_step_in_group?(estimate, step)
 
-      redirect_to wizard_path next_step_for(estimate, step)
+      # TODO: Stop saving data here
+      if Screens::TemporarySaveDeterminationService.call(estimate, step)
+        handler.save_data(cfe_connection, estimate_id, @form, session_data)
+      end
+
+      redirect_to wizard_path Screens::NextScreenNameService.call(estimate, step)
```

Comment:
I think this is the crux of the issue we are trying to solve. I think we should be thinking about data rather than screens, and also trying to find some cohesive objects somewhere inside the problem. The CFE spike raises the same time of question (how do we cope with 'save random blob of data from session into CFE?' in a way that it comprehensible)

Reply:
possibly the wizard gem has led us slightly down a blind alley here - it's thinking about knowing 'flow' between screens, and we can't know the answer in any reliable fashion